app/src/components/attachments/AttachmentsList.tsx

```diff
@@ -219,7 +221,20 @@ const AttachmentsList: React.FC<IAttachmentsListProps> = (props) => {
             <TableHead>
               <TableRow>
                 <TableCell width="60px" padding="checkbox">
-                  <Checkbox color="primary" />
+                  <Checkbox
+                    color="primary"
+                    onChange={(e) => {
+                      if (e.target.checked) {
+                        props.onCheckAllChange?.(
```

Comment:
change this to `if (e.target.checked && props.onCheckAllChange) { props.onCheckAllChange(...) }`